hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/predicate/IsNullTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.predicate;
+
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.ColumnExpression;
+import com.hazelcast.sql.impl.expression.ExpressionTestBase;
+import com.hazelcast.sql.impl.expression.SimpleExpressionEvalContext;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.apache.calcite.rel.type.RelDataType;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+
+import static com.hazelcast.sql.impl.calcite.validate.HazelcastSqlOperatorTable.IS_NULL;
+import static com.hazelcast.sql.impl.calcite.validate.types.HazelcastTypeSystem.narrowestTypeFor;
+import static org.apache.calcite.sql.type.SqlTypeName.BOOLEAN;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class IsNullTest extends ExpressionTestBase {
```

Comment:
One of the produced test cases:
```
query:  SELECT ? IS NULL FROM t
At line 1, column 8: Illegal use of dynamic parameter
```
What is the reason of this query to fail? It looks valid to me

Reply:
It's mostly for consistency with other expressions. Generally, if there is no enough type information to assign some meaningful types to operands, validation fails. Your idea is to assign `ANY` aka `OBJECT` type to `?` in this particular case of `? IS [NOT] NULL`, right?

Reply:
Yes, formally this is a valid expression.

Reply:
fixed in 310766b82df48678484150e207e0d14063465235